File: 187.重复的dna序列.go
/*
 * @lc app=leetcode.cn id=187 lang=golang
 *
 * [187] 重复的DNA序列
 */

// @lc code=start
func findRepeatedDnaSequences(s string) (res []string) {
	stringMap := map[string]int{}
	lengthOfStringNeedToCount := len(s) - 10
	for i := 0; i <= lengthOfStringNeedToCount; i++ {
		sub := s[i : i+10]
		stringMap[sub]++
		if stringMap[sub] == 2 {
			res = append(res, sub)
		}
	}
	return
}

// @lc code=end

File: 187.重复的dna序列.py
#
# @lc app=leetcode.cn id=187 lang=python3
#
# [187] 重复的DNA序列
#

# @lc code=start
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        string_dict = {}
        for i in range(len(s)-10 +1):
            if len(s)+1-i >= 10:
                if s[i:i+10] in string_dict:
                    string_dict[s[i:i+10]] += 1
                else:
                    string_dict[s[i:i+10]] = 1
        final_res = []
        for i,val in string_dict.items():
            if val > 1:
                final_res.append(i)
        return final_res
# @lc code=end

